### PR TITLE
Height of max

### PR DESCRIPTION
**Pull request by mspelman07**
This PR is superseeding PR: https://github.com/metoppv/improver/pull/2012 as I cannot push changes to the branch.
Acceptance_test_data: https://github.com/metoppv/improver_test_data/pull/51/files

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature

### CODE REVIEW

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Nice simple function and tests - easy to follow.

I haven't looked at the acceptance test data yet, but before I do, I think the units need fixing.

File: improver/utilities/cube_manipulation.py [hunk 1]
```diff
@@ -774,3 +774,52 @@ def maximum_in_height(
...
+        raise ValueError("Only high or low is valid.")
+    for height in height_points:
+        height_of_max.data = np.where(
+            cube[height].data == max_cube.data,
+            cube[height].coord("height").points[0],
+            height_of_max.data,
+        )
+    if new_name:
+        height_of_max.rename(new_name)
+    return height_of_max
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/utilities/cube_manipulation.py:826]: Done.

File: improver/utilities/cube_manipulation.py [hunk 2]
```diff
@@ -774,3 +774,52 @@ def maximum_in_height(
...
+    maximum value.
+
+    Args:
+        cube:
+            A cube with a height coordinate.
+        max_cube:
+            A cube of the maximum value over the height coordinate.
+        high_or_low:
+            Whether we are looking for the highest or lowest maximum height (for the case
+            where the maximum occurs at more than one height).
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/utilities/cube_manipulation.py]: Done. I've ended up doing the opposite of your suggestion and having find_lowest have a default value of True but I don't think this actually matters.

File: improver/utilities/cube_manipulation.py [hunk 3]
```diff
@@ -774,3 +774,52 @@ def maximum_in_height(
...
+        new_name:
+            The new name to be assigned to the output cube. If unspecified the name of the
+        original cube is used.
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/utilities/cube_manipulation.py]: Done

**Review by MoseleyS — APPROVED** [review]

**Pull request merged**